pkg/network/usm/utils/connection_rollup.go

```diff
@@ -11,169 +11,147 @@ import (
-// example below:
+// The approach here was designed such that it can be used in the context of a
+// *stream* of events, so we don't need to have the full data set in order to
+// aggregate different connections.
+//
+// Here's an input/output example:
 //
-// c1: (IP-A:60001, IP-B:8080)
-// c2: (IP-A:60002, IP-B:8080)
-// c3: (IP-B:8080, IP-A:60003)
+// |-------------------------------*-------------------------------|
+// | Input                         | Output                        |
+// |-------------------------------+-------------------------------|
+// | (1.1.1.1:60001 - 2.2.2.2:80)  | (1.1.1.1:60001 - 2.2.2.2:80)  |
+// | (1.1.1.1:60002 - 2.2.2.2:80)  | (1.1.1.1:60001 - 2.2.2.2:80)  |
+// | (1.1.1.1:60003 - 3.3.3.3:80)  | (1.1.1.1:60003 - 3.3.3.3:80)  |
+// | (1.1.1.1:60004 - 2.2.2.2:443) | (1.1.1.1:60004 - 2.2.2.2:443) |
+// | (1.1.1.1:60005 - 2.2.2.2:80)  | (1.1.1.1:60001 - 2.2.2.2:80)  |
+// | (1.1.1.1:60006 - 2.2.2.2:443) | (1.1.1.1:60004 - 2.2.2.2:443) |
+// | (2.2.2.2:80 - 1.1.1.1:70000)  | (2.2.2.2:80 - 1.1.1.1:60001)  |
+// | (2.2.2.2:90001 - 1.1.1.1:80)  | (2.2.2.2:90001 - 1.1.1.1:80)  |
+// | (2.2.2.2:90002 - 1.1.1.1:80)  | (2.2.2.2:90001 - 1.1.1.1:80)  |
+// |-------------------------------+-------------------------------|
 //
-// The approach is very simplistic and only aims to address the common cases we
-// see in most workloads. The function will likely generate "false negatives",
-// but ideally it shouldn't generate "false positives". In other words, we may
-// not always rollup two different `types.ConnectionKey` that may legitimately
-// refer to the same (client, server) pair, but we shouldn't cause correctness
-// issues by aggregating two datapoints that *don't* belong to to the same
-// (client, server) pair.
+// Note that the ephemeral port numbers just happen to be the number of the
+// first "sample" processed (eg. 60001, 60004, 90001) for a given
+// (IP-A,IP-B,SERVER_PORT) tuple.
 //
-// For the sake of documentation these are the edge-cases we *don't* intend to
-// address at this time:
+// It may be convenient for the client of this code to clear those ephemera
+// numbers and have something like (1.1.1.1:0 - 2.2.2.2:80) instead of
+// (1.1.1.1:60001 - 2.2.2.2:80) for the "rolled up" key.  For that purpose we
+// provided a second method (`ClearEphemeralPort`) that can be called once the
+// stream has been processed.
 //
-// * Bare process monitoring (this code shouldn't even be reached when bare
-// process monitoring is enabled);
-// * Gossip/p2p protocols where you see symmetrical traffic like
-// (IP-A:PORT-X,IP-B:PORT-Y) and (IP-B:PORT-X,IP-A:PORT-Y);
-// * Containers running multiple processes instances bound to different ports. A
-// real world example we have here at Datadog are redis containers running
-// multiple instances. In this case only one one of the redis instances would
-// trigger a rollup.
+// NOTE: this code should not be used in the context of bare-process monitoring.
```

Comment:
Are we truly not using it within the context of base-process monitoring?

Reply:
For now there is a FF disabling connection rollups by default, but we may need to "enforce" it programmatically if once we feel ready for enabling it by default. In any case I think it's fine to leave that for a future PR

Reply:
Can you please create Jira ticket for it so we wont forget about it?

Reply:
https://datadoghq.atlassian.net/browse/USMON-957